myuw/views/api/base_schedule.py

```diff
@@ -88,14 +89,12 @@ def load_schedule(request, schedule, summer_term=""):
             logger.error(ex)
             pass
 
-        if section.section_label() in canvas_data_by_course_id:
-            enrollment = canvas_data_by_course_id[section.section_label()]
-            # canvas_grade = enrollment.final_grade
-            # section_data["canvas_grade"] = canvas_grade
-            canvas_course = enrollment.course
-            if not canvas_course.is_unpublished():
-                section_data["canvas_url"] = canvas_course.course_url
-                section_data["canvas_name"] = canvas_course.name
+        try:
+            enrollment = canvas_enrollments[section.section_label()]
+            if canvas_course_is_available(enrollment.course_id):
```

Comment:
I think this needs to be in a thread per section.  Currently this could take timeout * 2 * number of course sections to finish.